world-client/src/networking/socket.js

```diff
@@ -37,13 +39,18 @@ export default class Socket {
     this.socket.onclose = (e) => {
       // TODO
     };
+
   };
 
   isConnected() {
     return this.socket && this.socket.readyState == WebSocket.OPEN;
   };
 
   send(msg) {
+    if(!this.isConnected()){
```

Comment:
Socket could still be in establishing connection state, when the first message is being sent. This will cause a reload, causing another slow handshake and another reload, causing an infinite (refresh) loop.
Handling should be aware of websocket.readyState and more focused on connection lost after it has already been established.

Reply:
I already thought of that with the `firstConnect` variable. Therefore this issue doesn't exist.